rfcs/0052-create-sink-into-table.md

```diff
@@ -0,0 +1,400 @@
+
+Currently, we can just support it with limitation that the mv's primary key can not be hidden column, we will discuss why in the "implementation" phrase.
+
+The statement requires **Allow a sink and a table have the same name**. It is ok in our system because the sink will never appear in the FROM clause. An alternative is that we can add a `_sink` suffix on the sink's name.
+
+## Implementation
+
+Thanks for @BugenZhao and @shanicky's offline discussion.
+#### CREATE SINK into TABLE
+
+Currently, the table's plan is shown as the graph. In brief, the table's data can comes from the external source or BatchDML. And the DMLExec is used here as a side input. ![](./images/0052-sink-to-table/cur-table-plan.png).
+
+The `create sink to table` introduces more data sources for the table. And now RW has implemented stream Union executor and RowIdGen with specified distribution. So we can support it more clearly with these new features.
+![](./images/0052-sink-to-table/new-table-plan.png)
+
+Here are some Q&A for the details of the graph
+- What's the difference between "append only table" and "table without pk"?
+  - The only difference is that if Update/delte on the table is allowed. which make the non-append only table without pk
+    - not append only, so the watermark is not allowed on it.
+    - the DML changes must be hash shuffled to the materialize fragment.
+    - the materialize must handle pk conflict for the DML changes.
+  - maybe we will support alter table between append only and non-append only in future.  
+- Why do we need blackhole SinkExecutor here? 
+  - we have some special logic in the sink executor such as `force-append-only` and special compaction when the down stream's pk does not contain stream key, we can reuse these logic in SinkExecutor to maintain the same behvaior with other kind of sinks.
+- "Exchange Any"
+  - It appears on the append only stream whose row_id column has not been generated. So any uniform shuffle strategy is ok here. 
+- Why not move the ProjectExec for the generated column on the Materialize's fragment?
+  - when there is a user defined pk on the table. The pk might contains generated column, which will be used to shuffle. So it must be generated before the exchange executor.
+  - when there is no pk, we can move it on the MaterializeExecutor's fragment. But I prefer make it consistent with the case having pk.
+- Default column Project
+  - For the Insert statement, the default column is handled in the batchInsert executor. To achieve the same behavior for the create sink into table, we must handle it between the sink and the materialize executor.
```

Comment:
Shall we change the implementation of `BatchInsert` to make them consistent?

By the way, what's the difference between `default` and `generated` column? 🤔